### PR TITLE
Remove HTML + PHP support, as must be handled in php-parser first

### PR DESCRIPTION
**Pull request by TomasVotruba**
This is one of design I want to solve before Rector 1.0 https://github.com/rectorphp/rector/issues/7854

The HTML and PHP support is now well grounded in php-parser - https://github.com/nikic/PHP-Parser/
We have bunch of workarounds for files, where Rector should not run at all. It only patches lack of types and does not provide much value. PHPStan does not know about the types and usage in HTML + PHP is limited, if any.

That's why we remove such hacks and rather skip any code with HTML + PHP nodes with warning. This will keep HTML files skipped without any touches and won't break them untill PHP is separated from HTML.

## Todo

- [ ] add warning on HTML node found

### CODE REVIEW

File: .github/workflows/e2e_with_cache.yaml [hunk 1]
```diff
@@ -25,7 +25,6 @@ jobs:
                 php_version: ['8.1']
                 directory:
                     - 'e2e/applied-rule-removed-node-with-cache'
-                    - 'e2e/timeout-file-not-cached'
```
Comment (samsonasik) [inline .github/workflows/e2e_with_cache.yaml]: This test is still needed.

**Review by samsonasik — CHANGES_REQUESTED** [review]
`e2e/timeout-file-not-cached` is still needed for test big file with limited time defined.

File: .github/workflows/e2e_with_cache.yaml (hunk 1 above)
Comment (TomasVotruba) [reply .github/workflows/e2e_with_cache.yaml]: I see. Now it depends on copy-paste printer of hundred lines.

Could you add a new test with just few lines, not with the printer, that does the same effect?

Comment (samsonasik) [reply .github/workflows/e2e_with_cache.yaml]: it can't only few lines, as it will succeed, which expect crash, it need a big file

Comment (samsonasik) [reply .github/workflows/e2e_with_cache.yaml]: I see, rector can be configured with 0 second:

```php
$rectorConfig->parallel(0);
```

for it.

Comment (samsonasik) [reply .github/workflows/e2e_with_cache.yaml]: I added it back with just few lines https://github.com/rectorphp/rector-src/pull/4051/commits/5a2cfeba698b36412b074fce63bb3b93f1ddbae7

**Comment by samsonasik** [comment]
@TomasVotruba I just realise rector build downgrade need php+html 

https://github.com/rectorphp/rector-src/blob/6dd57a913efe6fd5eb03542300b93cb1c22fc4dc/build/config/config-downgrade.php#L22

So it possibly break downgrade

**Comment by samsonasik** [comment]
See this PR for downgrade tracy files 

- https://github.com/rectorphp/rector-src/pull/3469

this is happen for immediate html+php patch, which on tracy part, probably fine, but will certainly break if append/prepend/remove stmt with InlineHTML as next or target refactor.

**Comment by samsonasik** [comment]
@leoloso FYI if you have mix php+html in your wordpress extension to downgrade

**Comment by samsonasik** [comment]
By add warning on found InlineHTML node, current downgrade will no longer work. The php+html hack was used on append/prepend/remove changed Stmt/InlineHTML (with file has changed check, that on purpose), not changed inside Stmt itself

**Comment by leoloso** [comment]
@samsonasik Thanks for the heads up. Luckily I don't have any mix of PHP + HTML to downgrade

**Comment by samsonasik** [comment]
@leoloso thank you for the check.

@TomasVotruba for handling summary, the issue was not about mix html+php, as inline change should works as expected, but when on target node near InlineHTML or after a Nop Stmt prepended/appended/removed, that InlineHTML need to be refreshed, or Nop need to be marked as comment. That's why the handling decorator uses start token check as new stmt doesn't have start token attribute yet.

**Comment by TomasVotruba** [comment]
@leoloso Hey, thanks for checking. Btw, how far down do you downgrade? 
I'm reviewing usefulness of target PHP versions, as some sets like PHP 7.0 or 7.1 are quite thin.

**Comment by leoloso** [comment]
@TomasVotruba I'm downgrading to PHP 7.1, and already very happy with this level. Some important WordPress plugins now require PHP 7.2, so I think nobody uses 7.0 anymore, and I certainly wouldn't spend much energy in adding the unsupported 7.0 features to the downgrade.

Btw, I believe that PHP 7.1 is fully covered with the downgrade, with all its features supported, so I'm not sure what you call "thin".

(Actually, I think that pretty much everything is supported between 7.1 and 8.0... only enums in 8.1 are the major feature not yet supported by the downgrade)

**Comment by TomasVotruba** [comment]
@samsonasik I'm working on this and rebase back and forth so I've lost your commit about timeout test. Could you add it via separate PR to dev-main so we have it? :)

**Comment by samsonasik** [comment]
@TomasVotruba yes https://github.com/rectorphp/rector-src/pull/4059

**Pull request merged**